src/scores/pandas/decorators.py

```diff
@@ -0,0 +1,78 @@
+"""
+Pandas scores decorators
+"""
+
+import functools
+from typing import Union
+
+import pandas as pd
+
+DATAFRAME_DOCS = """
+    Supports:
+        df: pd.DataFrame with column names specified in other keywords.
+            Set `{keywords}` to column names.
+            E.g.
+            >>> scores.pandas.METRIC_NAME(df, fcst = 'ForecastColumn', obs = 'ObsColumn')
+"""
+
+
+def split_dataframe(*series_keywords: str):
+    """
+    Allow for a `pd.DataFrame` to be provided, with `pd.Series` retrieved and passed to the underlying metric.
+
+    Searches the keyword arguments given to the metric for `series_keywords` to retrieve columns.
+
+    Expects the `pd.DataFrame` to either be the only argument, or under the `df` keyword.
+
+    Examples:
+        >>> @split_dataframe('fcst', 'obs')
+        >>> def mean_error(fcst, obs):
```

Comment:
Consider adding an example where you use the decorator to modify an existing metric (e.g., mse) with some column names that differ from the standard "fcst" and "obs"